src/unum/wireless/wireless_nl80211.c

```
@@ -0,0 +1,827 @@
+// (c) 2020 minim.co
+// unum wireless extensions APIs, used only by platforms
+// where drivers that work w/ nl80211 / mac80211 
+// Note: all the functions here are designed to be called from
+//       the wireless telemetry thread only
+
+#include "unum.h"
+#include "wireless_nl80211.h"
+
+/* Temporary, log to console from here */
+//#undef LOG_DST
+//#undef LOG_DBG_DST
+//#define LOG_DST LOG_DST_CONSOLE
+//#define LOG_DBG_DST LOG_DST_CONSOLE
+
+// Socket used to communicate with driver
+struct nl_sock *nl_sock = NULL;
+// NL80211 Identifier 
+int driver_id;
+struct handler_args {
+    const char *group;
+    int id;
+};
+
+// Differnt OUIs
+static unsigned char ms_oui[3]          = { 0x00, 0x50, 0xf2 };
+static unsigned char ieee80211_oui[3]   = { 0x00, 0x0f, 0xac };
+
+// Next index entry into the scan entries table
+int scan_index = 0;
+
+// Init NL80211 socket to communicate with NL80211 / WLAN driver
+static int wt_nl80211_init()
+{
+    int err;
+
+    nl_sock = nl_socket_alloc();
+    if (!nl_sock) {
+        log("Failed to allocate netlink socket.\n");
+        return -1;
+    }
+
+    if (genl_connect(nl_sock)) {
+        log("%s: Failed to connect to generic netlink.\n", __func__);
+        nl_socket_free(nl_sock);
+        return -2;
+    }
+
+    nl_socket_set_buffer_size(nl_sock, 8192, 8192);
+
+    /* try to set NETLINK_EXT_ACK to 1, ignoring errors */
+    err = 1;
+    setsockopt(nl_socket_get_fd(nl_sock), SOL_NETLINK,
+            NETLINK_EXT_ACK, &err, sizeof(err));
```

It might be useful to log message here if it fails (even if the error is ignored). That's assuming extended ACK is used anywhere in the code... If not then no point to even try enabling it.

---

Added a log message.